### PR TITLE
Add additional logging. Remove unnecessary router lookups. May addres…

### PR DESCRIPTION
**Pull request by plorenz**
…s #1460

### CODE REVIEW

File: controller/handler_ctrl/accept.go [hunk 1]
```diff
@@ -59,64 +59,75 @@ func (self *CtrlAccepter) Bind(binding channel.Binding) error {
...
-							WithField("costTags", listener.GetCostTags()).
-							Debug("router listener")
-					}
-				}
+	r, err := self.network.GetReloadedRouter(ch.Id())
+	if err != nil {
+		return err
+	}
+	if r == nil {
+		return errors.Errorf("no router with id [%v] found, closing connection", ch.Id())
```
Comment (dovholuknf) [inline controller/handler_ctrl/accept.go:70]: "closing connection" is not actually happening here and is misleading imo.

File: controller/handler_ctrl/accept.go [hunk 2]
```diff
@@ -59,64 +59,75 @@ func (self *CtrlAccepter) Bind(binding channel.Binding) error {
...
 		} else {
 			return errors.New("no version info header, closing router connection")
 		}
 
-		r.Control = ch
-		r.ConnectTime = time.Now()
-		if err := binding.Bind(newBindHandler(self.heartbeatOptions, r, self.network, self.xctrls)); err != nil {
-			return errors.Wrap(err, "error binding router")
+		r.Listeners = nil
+		if val, found := ch.Underlay().Headers()[int32(ctrl_pb.ContentType_ListenersHeader)]; found {
```
[the remark this answers is not in the document]
Comment (dovholuknf) [reply controller/handler_ctrl/accept.go:90]: (i also see now that is how it always was, originally i thought tihs was new code so... nevermind prolly)

File: controller/handler_ctrl/accept.go [hunk 3]
```diff
@@ -59,64 +59,75 @@ func (self *CtrlAccepter) Bind(binding channel.Binding) error {
...
+				r.SetLinkListeners(listeners.Listeners)
+				for _, listener := range listeners.Listeners {
+					log.WithField("address", listener.GetAddress()).
+						WithField("protocol", listener.GetProtocol()).
+						WithField("costTags", listener.GetCostTags()).
+						Debug("router listener")
+				}
+			}
+		} else {
+			log.Info("no advertised listeners")
```
Comment (dovholuknf) [inline controller/handler_ctrl/accept.go:104]: is this really an info type of message?

File: controller/handler_ctrl/accept.go [hunk 4]
```diff
@@ -59,64 +59,75 @@ func (self *CtrlAccepter) Bind(binding channel.Binding) error {
...
+	if err := binding.Bind(newBindHandler(self.heartbeatOptions, r, self.network, self.xctrls)); err != nil {
+		return errors.Wrap(err, "error binding router")
+	}
 
-		self.network.ConnectRouter(r)
+	if self.traceHandler != nil {
+		binding.AddPeekHandler(self.traceHandler)
 	}
+
+	log.Info("accepted new router connection")
```
Comment (dovholuknf) [inline controller/handler_ctrl/accept.go:128]: why lose the existing, presumably useful, `r.Id`? is the `[r/Id]` not useful?

**Review by dovholuknf — APPROVED** [review]
nothing looks like it shoudl stop the pr, just some questions here/there

File: controller/handler_ctrl/accept.go (hunk 1 above)
Comment (plorenz) [reply controller/handler_ctrl/accept.go:70]: changed to 'not accepting' so it's more accurate.

File: controller/handler_ctrl/accept.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (plorenz) [reply controller/handler_ctrl/accept.go:90]: ctrl_pb.ContentType_ListenersHeader is a ContentType which is an alias for an int32, so the cast is fine.

File: controller/handler_ctrl/accept.go (hunk 4 above)
Comment (plorenz) [reply controller/handler_ctrl/accept.go:128]: There's already a logger field for routerId added at the top, so this was redundant.

File: controller/handler_ctrl/accept.go (hunk 3 above)
Comment (plorenz) [reply controller/handler_ctrl/accept.go]: It was Warn and I dropped it to Info. Dropped it further to Debug.

**Pull request merged**